### PR TITLE
[DEV-6210] Remove Dead JS, Upgrade Webpack + Node

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**

Upgrades webpack and node.

image

**Technical details:**
see commits!!

**JIRA Ticket:**
[DEV-6210](https://federal-spending-transparency.atlassian.net/browse/DEV-6210)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
`N/A` Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
`N/A` Verified mobile/tablet/desktop/monitor responsiveness
`N/A` Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
`N/A` Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
`N/A` [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
`N/A` [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`
- [x] Upgrade node in DockerFile and `.node-version`

Reviewer(s):
`N/A` Design review complete `if applicable`
`N/A` [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete
- [ ] Tested in sandbox for breaking changes + performance (pull down basic auth and dynamic renderer for performance test)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,12 +15,9 @@ env:
 cache:
   directories:
     - node_modules
-before_install:
-  - npm install -g @lhci/cli@0.4.x
```
Comment (jonhill13) [inline .travis.yml:19]: 👍

File: package.json [hunk 2]
```diff
@@ -1,137 +1,129 @@
...
-        "tunnel": "^0.0.6",
-        "xmldoc": "^0.4.0"
-    }
+  "name": "usaspending-website",
+  "version": "2.5.0",
+  "description": "USAspending Website",
+  "repository": "fedspendingtransparency/usaspending-website",
+  "main": "src/index.js",
+  "scripts": {
+    "start": "npx webpack serve --hot-only  --open --history-api-fallback --progress --config ./webpack/webpack.dev.config.js",
```
Comment (jonhill13) [inline package.json:8]: looks like there are two spaces between --hot-only and --open

File: package.json [hunk 3]
```diff
@@ -1,137 +1,129 @@
...
+    "lint": "npx eslint --config .eslintrc --ext .jsx,.js \"src/js/**\"; exit 0;",
+    "errors": "npx eslint --config .eslintrc --ext .jsx,.js \"src/js/**\" --quiet; exit 0;",
+    "dev": "npx webpack --progress --config ./webpack/webpack.dev.config.js",
+    "prod": "npx webpack --progress --config ./webpack/webpack.prod.config.js",
+    "travis": "npm audit --audit-level=moderate --production && npx eslint --config .eslintrc --ext .jsx,.js \"src/js/**\" && npx jest --runInBand;",
+    "light-house": "npx webpack --progress --config ./webpack/webpack.prod.config.js ; npx lhci autorun",
+    "build-redirect-config": "npx babel-node scripts/buildRedirectConfig.js"
+  },
+  "author": "",
+  "engines": {
```
Comment (jonhill13) [inline package.json:20]: do we want to update these engines?

File: src/index.ejs [hunk 4]
```diff
@@ -20,7 +20,7 @@
     <script async type="text/javascript" id="_fed_an_ua_tag"
         src="https://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js?agency=TREAS&subagency=FS"></script>
     
-    <script src='https://api.mapbox.com/mapbox-gl-js/v1.13.0/mapbox-gl.js'></script>
+    <script src='https://api.mapbox.com/mapbox-gl-js/v1.13.0/mapbox-gl.js' async></script>
```
Comment (jonhill13) [inline src/index.ejs:23]: too legit

File: webpack/webpack.common.js [hunk 5]
```diff
@@ -25,31 +27,14 @@ module.exports = {
...
-                    name: 'common',
-                    minChunks: 2,
-                    chunks: 'all',
-                    priority: 10,
-                    reuseExistingChunk: true,
-                    enforce: true
-                }
-            }
-        }
+        moduleIds: 'deterministic'
```
Comment (jonhill13) [inline webpack/webpack.common.js:37]: 🔥

File: webpack/webpack.prod.config.js [hunk 6]
```diff
@@ -16,27 +15,13 @@ module.exports = merge(common, {
...
-                styles: {
-                    // all css in one file -- https://github.com/webpack-contrib/mini-css-extract-plugin
-                    name: "styles",
-                    test: /\.css$/,
-                    chunks: "all",
-                    enforce: true
-                }
-            }
-        }
+            new CssMinimizerPlugin()
```
Comment (jonhill13) [inline webpack/webpack.prod.config.js:23]: 🔥

**Review by jonhill13 — DISMISSED** [review]

**Comment by jonhill13** [comment]
image
did you notice this warning and error, is there any way to resolve these?

**Comment by MaxwellKendall** [comment]
image

File: package.json (hunk 3 above)
Comment (MaxwellKendall) [reply package.json:20]: yes

File: package.json (hunk 2 above)
Comment (MaxwellKendall) [reply package.json:8]: ah thank you sir

File: webpack/webpack.common.js [hunk 7]
```diff
@@ -3,9 +3,11 @@ const path = require('path');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
 const CleanWebpackPlugin = require('clean-webpack-plugin');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
-const GitRevisionPlugin = require('git-revision-webpack-plugin');
+const GitRevisionPlugin = require('git-revision-webpack-plugin').GitRevisionPlugin;
 const CopyWebpackPlugin = require('copy-webpack-plugin');
+const MomentLocalesPlugin = require('moment-locales-webpack-plugin');
 
+const isDevelopment = process.env.NODE_ENV === 'development';
```
Comment (ebdabbs) [inline webpack/webpack.common.js:10]: Does this need to change now that we've gotten rid of development env?

**Review by ebdabbs — CHANGES_REQUESTED** [review]
I'm also getting
```
WARNING in DefinePlugin
Conflicting values for 'process.env'
```
when running the app without modifications to GlobalConstants

**Comment by ebdabbs** [comment]
Closing in favor of #2611 / completing this work in smaller increments. This branch should still be a good reference.

**Pull request closed without merging**